### PR TITLE
PB-5488: Change logic to check is original backups are synced or not

### PR DESCRIPTION
**Pull request by kshithijiyer-px**
**What this PR does / why we need it**:
- [x] Change logic to check is original backups are synced or not

Dashbaord URL: 
https://aetos.pwx.purestorage.com/resultSet/testSetID/531787
https://aetos.pwx.purestorage.com/resultSet/testSetID/531826

**Which issue(s) this PR fixes** (optional)
Closes #PB-5488

**Special notes for your reviewer**:

### CODE REVIEW

**Review by snigdha-px — APPROVED** [review]
LGTM

**Review by atrivedi-px — APPROVED** [review]
LGTM

**Review by sn-px — APPROVED** [review]
LGTM

**Review by snigdha-px — APPROVED** [review]

File: tests/backup/backup_restore_basic_test.go [hunk 1]
```diff
@@ -1297,7 +1298,14 @@ var _ = Describe("{BackupSyncBasicTest}", func() {
 			log.FailOnError(err, "Wait for BackupSync to complete")
 			fetchedBackupNames, err := GetAllBackupsAdmin()
 			log.FailOnError(err, "Getting a list of all backups")
-			dash.VerifyFatal(len(fetchedBackupNames), len(backupNames), "Comparing the expected and actual number of backups")
+
+			// Iterating through all backup and check if they are present in the fetched backup list or not
+			listOfFetchedBackups := strings.Join(fetchedBackupNames, "")
+			for _, backup := range backupNames {
+				re := regexp.MustCompile(fmt.Sprintf("%s-*", backup))
```
Comment (kphalgun-px) [inline tests/backup/backup_restore_basic_test.go:1305]: Can we use IsPresent or a map to enhance readability or efficiency?


```go
for _, backup := range backupNames {
    found := IsPresent(fetchedBackupsNames, backup)
    dash.VerifyFatal(found, true, fmt.Sprintf("Checking if backup [%s] was synced or not", backup))
}
```

```go
fetchedBackupsMap := make(map[string]struct{}) // struct{} occupies 0 bytes
for _, name := range fetchedBackupNames {
    fetchedBackupsMap[name] = struct{}{}
}

for _, backup := range backupNames {
    _, found := fetchedBackupsMap[backup]
    dash.VerifyFatal(found, true, fmt.Sprintf("Checking if backup [%s] was synced or not", backup))
}
```

**Review by mkoppal-px — APPROVED** [review]
LGTM

File: tests/backup/backup_restore_basic_test.go (hunk 1 above)
Comment (kshithijiyer-px) [reply tests/backup/backup_restore_basic_test.go:1305]: No @kphalgun-px, I tried that earlier the problem with that is when you sync the backup it's getting appended with an hash value which is random and if you go with the `IsPresent()` call it tries to match it exactly.

Comment (kphalgun-px) [reply tests/backup/backup_restore_basic_test.go:1305]: Thanks for clarification.

Comment (kshithijiyer-px) [reply tests/backup/backup_restore_basic_test.go:1305]: Reference link of the job where I tired this approach https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/px-backup-on-demand-system-test-byoc/4241/console

**Review by sn-px — APPROVED** [review]
LGTM

**Review by atrivedi-px — APPROVED** [review]

**Pull request merged**